### ISSUE TITLE
[Accessibility] NVDA tool does not show focus blue rectangle in property edit

### ISSUE THREAD

**Issue by MelonWang1** [dotnet/winforms#12565]
### .NET version

.NET 10.0.100-alpha.1.24573.1

### Did it work in .NET Framework?

Yes

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, also can repro in .NET8.0/9.0, cannot repro in .NET 6.0/7.0.

## Current Behavior

NVDA does not show focus blue rectangle.
![Image](https://github.com/user-attachments/assets/90c83755-bc2b-4d32-9318-c94bb037fb2a)

https://github.com/user-attachments/assets/a147db27-d749-4819-840f-d11e26e5aa45

## Expected Behavior

NVDA should show focus blue rectangle.
In .NET7.0/6.0:
![Image](https://github.com/user-attachments/assets/89c97c0a-3093-4f5a-b4a9-976c337f3b72)

https://github.com/user-attachments/assets/f56a1fbe-6a29-4320-b404-cc39df126aa1

In .NET Framework:

![Image](https://github.com/user-attachments/assets/1dd6a110-90ef-4777-9475-96d88b31f23d)

### Steps to reproduce

1. Create a .NET Core Winforms project.
2. Add a Button and PropertyGrid controls in form designer.
3. Set the SelectedObject property to Button1 in propertyGrid1 control.
4. Build and run project.
5. Open NVDA tool and select FlatStyle property in propertyGrid control.
6. Press Tab (should focus on edit part)

**Comment by LeafShi1** [comment]
@MelonWang1 Could you please test if the issue is related to the version of NAVD and confirm if the issue occurs in Narrator or JAWS?

**Comment by MelonWang1** [comment]
@LeafShi1 This issue cannot repro in Narrator tool, can repro in latest NVDA tool version 2024.4.1.
Narrator:
![Image](https://github.com/user-attachments/assets/876ad93f-b04a-4c31-9d21-c20c9109cb19)

NVDA:
![Image](https://github.com/user-attachments/assets/d6e87e41-637d-4662-a702-146a8ffa08fe)

**Comment by LeafShi1** [comment]
This issue caused by the changes in [PR #9321](https://github.com/dotnet/winforms/pull/9321)

**Comment by LeafShi1** [comment]
This issue seems to be because the MSAA event is not triggered, resulting in the GridViewTextBox's state cannot be read by NVDA. 
![Image](https://github.com/user-attachments/assets/497be097-04db-4de1-812f-ed3084c0107b)


I tried to add the event `AccessibilityNotifyClients(AccessibleEvents.Focus, 0, -1); ` in method `GridViewTextBox.OnGotFocus()`, but the following exception is thrown during the code running


<img width="713" alt="Image" src="https://github.com/user-attachments/assets/9bfa80ac-c208-4e16-8502-6e525ef9db10" />


@ricardobossan Please confirm whether this exception is caused by the change of https://github.com/dotnet/winforms/pull/9321

**Comment by ricardobossan** [comment]
@LeafShi1 I checked out the commit right before PR #9321 and added the following line to the `GridViewTextBox.OnGotFocus()` method:

```csharp
AccessibilityNotifyClients(AccessibleEvents.Focus, 0, -1);
```

After adding this, I tested the reproduction steps while debugging. However, no error was thrown, and the issue did not reproduce for me.

![Image](https://github.com/user-attachments/assets/431d24a0-036f-465a-803c-baf10c449c80)

**Comment by chaowendy** [comment]
Verified this issue in the latest .NET 10.0 SDK build: 10.0.100-alpha.1.24623.6 + private dlls built from the main branch of Winforms repo, now NVDA show focus blue rectangle in property edit.
![Image](https://github.com/user-attachments/assets/6e45f51d-ae45-4646-bb61-e04b79c480bc)


But there is a regression issue that each option will be read three times, and from alphanumeric long to short the last time will be announced extra word "**instead**", for example from 'Flat' to 'System', will be read once "**System selected instead**".
![Image](https://github.com/user-attachments/assets/87d04521-0be2-4695-8491-2584b822997d)


**which is only read once before.** 
![Image](https://github.com/user-attachments/assets/48b84036-9247-4135-99e7-76d9f673bff1)

**Comment by LeafShi1** [comment]
The last time announced extra word "instead" is caused by the code
https://github.com/dotnet/winforms/blob/49c20e50c741801f50217a23d2eb6d0f1e860e98/src/System.Windows.Forms/src/System/Windows/Forms/Controls/TextBox/TextBoxBase.cs#L1732

**Comment by LeafShi1** [comment]
@JeremyKuhne Can you take a look at this issue, it is a regression caused by PR #9321

After I added MSAA events(PR #12651) to the control, the NVDA rectangle can focus on the property edit box, but before PR #9321, we don't need to add this extra MSAA event.  And adding this event will cause NVDA to read the edit box information three times.

**Comment by LeafShi1** [comment]
The root cause in [Control.ControlAccessibleObject.cs](https://github.com/dotnet/winforms/blob/d1bbb322176109d0ace6b2de2e5f578a94ad06a9/src/System.Windows.Forms/src/System/Windows/Forms/Accessibility/Control.ControlAccessibleObject.cs#L470-L474) 

<img width="539" alt="Image" src="https://github.com/user-attachments/assets/20d69ace-64be-4c70-930e-2834d20342a7" />

**Comment by merriemcgaw** [comment]
@LeafShi1  @MelonWang1 is this an NVDA only issue? Can you show me what happens with Accessibility Insights for Windows? If we have the correct things showing there, we can close this bug as external.

**Comment by MelonWang1** [comment]
@LeafShi1 @merriemcgaw Yes, this issue cannot repro in Accessibility Insights tool.  

For now, in latest .NET SDK 10.0.100-preview.3.25160.26, the NVDA does not show focus blue rectangle in property edit. It is not consistent with[ PR12651](https://github.com/dotnet/winforms/pull/12651) fix and [verify](https://github.com/dotnet/winforms/issues/12565#issuecomment-2562377687) result. Is this expected?

**Comment by LeafShi1** [comment]
> [@LeafShi1](https://github.com/LeafShi1) [@merriemcgaw](https://github.com/merriemcgaw) Yes, this issue cannot repro in Accessibility Insights tool.
> 
> For now, in latest .NET SDK 10.0.100-preview.3.25160.26, the NVDA does not show focus blue rectangle in property edit. It is not consistent with[ PR12651](https://github.com/dotnet/winforms/pull/12651) fix and [verify](https://github.com/dotnet/winforms/issues/12565#issuecomment-2562377687) result. Is this expected?

It is indeed as you said, in latest .NET SDK, the previously added MSAA events did not fix the original issue.

@merriemcgaw  Please confirm whether we need to investigate this issue further?

**Comment by merriemcgaw** [comment]
Given the investigation and that this is only NVDA, I think we can close this issue.
